### ISSUE TITLE
Re-use of fitted submodels

### ISSUE THREAD

**Issue by fweber144** [stan-dev/projpred#168]
Argument `cv_search` currently seems to be be responsible for fetching submodels already fitted during a search (see issue #154 and function `.get_submodels()`). The relevant code seems to be line https://github.com/stan-dev/projpred/blob/16770099ae45c24a3e9b1798ec46c36218314a68/R/projfun.R#L66 There however, the fitted submodels from `search_path` are fetched which usually have a coarser reference distribution (`p_sel`) than used for prediction (`p_pred`), see lines https://github.com/stan-dev/projpred/blob/16770099ae45c24a3e9b1798ec46c36218314a68/R/varsel.R#L96-L98 and https://github.com/stan-dev/projpred/blob/16770099ae45c24a3e9b1798ec46c36218314a68/R/cv_varsel.R#L292-L304 and https://github.com/stan-dev/projpred/blob/16770099ae45c24a3e9b1798ec46c36218314a68/R/cv_varsel.R#L558-L559 where `ndraws` and `nclusters` are usually smaller than their `_pred` variants. So my question is: Wouldn't it make more sense to let `cv_search` fetch the fitted submodels which are based on `p_pred`, i.e., those which were used for prediction? Of course, that would only be possible when using `cv_search` in the context of `project()` (not within `varsel()` or `cv_varsel()`). In order to be able to do this, I guess `.get_submodels()` would need a new (optional) argument accepting the fitted submodels based on `p_pred`. Furthermore, `p_sub` from https://github.com/stan-dev/projpred/blob/16770099ae45c24a3e9b1798ec46c36218314a68/R/varsel.R#L110-L113 would need to be added to the respective `"vsel"` output object (which is probably helpful anyway) of `varsel()` and `cv_varsel()`. Then `project()` could pass that `p_sub` component of the `"vsel"` object to `.get_submodels()`.

**Comment by AlejandroCatalina** [comment]
So the idea of this argument is to avoid computational cost by not refitting the projections with `ndraws_pred` precisely. We can add your suggestion for `project` only. By the way, this argument will go back to being named `relax` probably, as it matches its purpose properly.

**Comment by fweber144** [comment]
> So the idea of this argument is to avoid computational cost by not refitting the projections with ndraws_pred precisely.

Yes, my proposition would also avoid refitting with `ndraws_pred` as it would re-use the corresponding fits from `varsel()` or `cv_varsel()` in `project()`.

> We can add your suggestion for project only.

Right, I mentioned that in my original comment, too. But I admit that I could have pointed it out more strongly.

Thanks for taking this issue into consideration. It's not urgent, but might be worth to be implemented in the future.

**Comment by fweber144** [comment]
While re-reading this, I stumbled across your statement

> By the way, this argument will go back to being named `relax` probably, as it matches its purpose properly.

You mean something like `reproject` (or `refit_proj` which occurs on branch `workflow`) and not `relax`, right? In #154, @jpiironen said the excerpt of the documentation I quoted there came from an argument called `relax`, but that only matches the job of `cv_search`  in case of L1 search. Now (where `cv_search` can also be applied to forward search), the job of `cv_search` is more accurately described by `reproject` or `refit_proj`, I think.
